src/main/java/reposense/git/GitVersion.java

```diff
@@ -4,19 +4,20 @@
 
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 /**
  * Contains git version related functionalities.
  * Git version is responsible for finding out the version of git the user of RepoSense is running.
  */
 public class GitVersion {
-    public static final String FINDING_PREVIOUS_AUTHORS_INVALID_VERSION_WARNING_MESSAGE =
-            "--find-previous-authors/-F requires git version 2.23 and above. Feature will be disabled for this run";
 
-    /** Regex for matching Git version 2.23 and above */
-    public static final Pattern FINDING_PREVIOUS_AUTHORS_VALID_GIT_VERSION_PATTERN =
-            Pattern.compile("(((2\\d*\\.(2[3-9]\\d*|[3-9]\\d+|\\d{3,}))|((([3-9]\\d*)|(1\\d+))\\.\\d+))\\.\\d*)");
+    /** Regex for finding Git Version */
```

Comment:
The original one simply checks if the version is `2.23` and above. Am I right to say that new one is trying to have a more general method that checks if the version is above this input version (in this case `2.23`) that we choose?

Reply:
The new one roughly implements general version comparison. I'm actually not too sure what the original pattern finds, which is my general issue with using extensive regex patterns. I believe even the first part `2\\d*\\.`  is looking for git versions of the form `2.`, `20.` `211.` and so on but I believe fails for even something like git version `1.5.0` or `3.0.0`. 

There is a `Version` class introduced in Java 9 but we're still on Java 8.

Reply:
Thanks for the answer. Agree that this will be helpful in future if we need another version checks!